mne/decoding/tests/test_transformer.py

```diff
@@ -39,26 +39,46 @@ def test_scaler():
     epochs = Epochs(raw, events, event_id, tmin, tmax, picks=picks,
                     baseline=(None, 0), preload=True)
     epochs_data = epochs.get_data()
-    scaler = Scaler(epochs.info)
     y = epochs.events[:, -1]
 
-    X = scaler.fit_transform(epochs_data, y)
-    assert_true(X.shape == epochs_data.shape)
-    X2 = scaler.fit(epochs_data, y).transform(epochs_data)
-    assert_array_equal(X2, X)
-    # these should be across time
-    assert_allclose(X.std(axis=-2), 1.)
-    assert_allclose(X.mean(axis=-2), 0., atol=1e-12)
-
-    # Test inverse_transform
-    Xi = scaler.inverse_transform(X, y)
-    assert_array_almost_equal(epochs_data, Xi)
-
-    for kwargs in [{'with_mean': False}, {'with_std': False}]:
-        scaler = Scaler(epochs.info, **kwargs)
-        scaler.fit(epochs_data, y)
-        assert_array_almost_equal(
-            X, scaler.inverse_transform(scaler.transform(X)))
+    methods = (None, dict(mag=5, grad=10, eeg=20), 'mean', 'median')
+    infos = (epochs.info, epochs.info, None, None)
+    epochs_data_t = epochs_data.transpose([1, 0, 2])
+    for method, info in zip(methods, infos):
+        if method == 'median' and not check_version('sklearn', '0.17'):
+            assert_raises(ValueError, Scaler, info, method)
+            continue
+        if method == 'mean' and not check_version('sklearn', ''):
+            assert_raises(ImportError, Scaler, info, method)
+            continue
+        scaler = Scaler(info, method)
+        X = scaler.fit_transform(epochs_data, y)
+        assert_equal(X.shape, epochs_data.shape)
+        if method is None or isinstance(method, dict):
+            sd = DEFAULTS['scalings'] if method is None else method
+            stds = np.zeros(len(picks))
```

Comment:
I don't understand, `int` has no len(). But that's probably not worth a discussion..